src/executor/create_executor.cpp

```diff
@@ -111,6 +110,14 @@ bool CreateExecutor::CreateTable(const planner::CreatePlan &node) {
   if (current_txn->GetResult() == ResultType::SUCCESS) {
     LOG_TRACE("Creating table succeeded!");
 
+    // Initialize table lock
+    auto table_object = catalog::Catalog::GetInstance()->GetTableObject(
+        database_name, table_name, current_txn);
+
+    oid_t table_oid = table_object->GetTableOid();
+    concurrency::LockManager *lm = concurrency::LockManager::GetInstance();
+    lm->InitLock(table_oid, concurrency::LockManager::RW_LOCK);
```

Comment:
Should we handle negative case here? What if InitLock is failed.

Reply:
Currently it won't do anything, but yes I agree with you.